apps/prairielearn/src/pages/studentGradebook/studentGradebook.ts

```diff
@@ -23,12 +26,55 @@ function buildCsvFilename(locals: Record<string, any>) {
   return courseInstanceFilenamePrefix(locals.course_instance, locals.course) + 'gradebook.csv';
 }
 
+const StudentGradebookRowSchema = z.object({
+  assessment: StudentAssessmentSchema,
+  assessment_instance: StudentAssessmentInstanceSchema,
+  assessment_set: StudentAssessmentSetSchema,
+  show_closed_assessment_score: z.boolean(),
+});
+
+type StudentGradebookRowRaw = z.infer<typeof StudentGradebookRowSchema>;
+
+function computeTitle({ assessment, assessment_instance }: StudentGradebookRowRaw) {
+  if (assessment.multiple_instance) {
+    return `${assessment.title} instance #${assessment_instance.number}`;
+  }
+  return assessment.title ?? '';
+}
+
+function computeLabel({ assessment, assessment_instance, assessment_set }: StudentGradebookRowRaw) {
+  if (assessment.multiple_instance) {
+    return `${assessment_set.abbreviation}${assessment.number}#${assessment_instance.number}`;
+  }
+  return `${assessment_set.abbreviation}${assessment.number}`;
+}
+
+function mapRow(
+  raw: StudentGradebookRowRaw,
+  prev: StudentGradebookRowRaw | null,
+): StudentGradebookRow {
+  // true if this is the first row or assessment_set.id differs from previous
```

Comment:
Nit: this reads as a verbatim description of the next line, is it needed?

Reply:
Probably not, removed in f729cd7b2d640e436926d5be083d3866efee4fc7